src/dsnotificationinterface.cpp

```diff
@@ -79,7 +79,7 @@ void CDSNotificationInterface::UpdatedBlockTip(const CBlockIndex *pindexNew, con
     if (pindexNew == pindexFork) // blocks were disconnected without any new ones
         return;
 
-    m_mn_sync.UpdatedBlockTip(m_chainman.m_best_header, pindexNew, fInitialDownload);
+    m_mn_sync.UpdatedBlockTip(WITH_LOCK(::cs_main, return m_chainman.m_best_header), pindexNew, fInitialDownload);
```

Comment:
I guess we should add EXCLUSIVE_LOCKS_REQUIRED to this variable then

Reply:
it is added by `bitcoin/bitcoin#26140` which is far ahead and have many unrelated changes